objective.py

```diff
@@ -6,54 +6,127 @@
 with safe_import_context() as import_ctx:
     from sklearn.dummy import DummyClassifier
     from sklearn.model_selection import train_test_split
-
+    from sklearn.metrics import balanced_accuracy_score as BAS
+    import numpy as np
+    from skorch.helper import SliceDataset, to_numpy
 
 # The benchmark objective must be named `Objective` and
 # inherit from `BaseObjective` for `benchopt` to work properly.
+
+
 class Objective(BaseObjective):
 
     # Name to select the objective in the CLI and to display the results.
     name = "BCI"
 
-    requirements = ["scikit-learn"]
-
     # List of parameters for the objective. The benchmark will consider
     # the cross product for each key in the dictionary.
     # All parameters 'p' defined here are available as 'self.p'.
-    # This means the OLS objective will have a parameter `self.whiten_y`.
+
+    link = 'pip: git+https://github.com/Neurotechx/moabb@develop#egg=moabb'
+    intall_cmd = 'conda'
+    requirements = [str(link),
+                    'scikit-learn']
+
     parameters = {
-        'seed': [42],
+        'evaluation_process, subject, subject_test, session_test': [
+            ('intra_subject', 1, None, None),
+             ],
     }
-
+    # The solvers will train on all the subject except subject_test.
+    # It will be the same for the sessions.
     # Minimal version of benchopt required to run this benchmark.
     # Bump it up if the benchmark depends on a new feature of benchopt.
     min_benchopt_version = "1.3.2"
 
-    def set_data(self, X, y):
+    def set_data(self, dataset, paradigm_name, sfreq):
```

Comment:
why do you need `sfreq`? can't you recover it from the dataset?

Reply:
Not sure I understand this point: isn't `dataset` a braindecode object?